test/functional/shared/cli/cli_test.go

```diff
@@ -413,6 +414,67 @@ func Test_Run_Portforward(t *testing.T) {
 	require.Equal(t, err, ctx.Err(), "rad run should have been canceled")
 }
 
+func Test_Run_RadInit(t *testing.T) {
+	ctx, cancel := testcontext.NewWithCancel(t)
+	t.Cleanup(cancel)
+
+	options := shared.NewRPTestOptions(t)
+	cli := radcli.NewCLI(t, options.ConfigFilePath)
+
+	args := []string{
+		"init",
+	}
+
+	// rad init will prompt for input, so we need to pipe in the answers.
+	cmd, heartbeat, description := cli.CreateCommand(ctx, args)
+
+	// Read from stdout to get the logs.
+	stdout, err := cmd.StdoutPipe()
+	require.NoError(t, err)
+
+	f, err := pty.Start(cmd)
+	require.NoError(t, err)
+
+	// Write to stdin to provide input.
+	go func() {
+		f.Write([]byte("j"))
+		f.Write([]byte(" "))
+	}()
+
+	// Start heartbeat to trigger logging
+	done := make(chan struct{})
+	go heartbeat(done)
+
+	// Read the text line-by-line while the command is running, but store it so we can report failures.
+	output := bytes.Buffer{}
+	scanner := bufio.NewScanner(stdout)
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		output.WriteString(line)
+		output.WriteString("\n")
+	}
+
```

Comment:
Some thoughts about this:

This test has a side-affect on the world. It installs Radius and sets up a new environment. I think if we really want to test `rad init` then we'd run this on a clean Kubernetes cluster in its own **separate** test run. 

I'm not yet convinced that this is helping us solve a problem. How would we use this to test something more complex like setting up AWS credentials?

Reply:
1. I agree with the first point. It should have its own Kubernetes cluster, otherwise, it might affect ongoing tests.
2. If we know the selections that we need to make while running a command we can give those selections as inputs to the test and it is going to do what we ask it to do.

In this case, I want to test `rad init` and I would like the command to not create an application so I want to select **No** when prompted. This is going to run `rad init` for me and then I will be able to query the dev recipes. This would make sure that nothing breaks our `rad init` happy path and can be extended to a lot of different scenarios.

Let me know if you have a few minutes so we can discuss this offline @rynowak.